skillmap/src/App.tsx

```diff
@@ -222,72 +226,62 @@ class AppImpl extends React.Component<AppProps, AppState> {
 
             let currentUser = await getUserStateAsync();
             let headerIds = getFlattenedHeaderIds(localUser, state.pageSourceUrl, currentUser);
-
             // Tell the editor to transfer local skillmap projects to the cloud.
             const headerMap = (await res.sendMessageAsync!({
                 type: "pxteditor",
                 action: "savelocalprojectstocloud",
                 headerIds
             } as pxt.editor.EditorMessageSaveLocalProjectsToCloud)).resp.headerIdMap;
-
             if (headerMap) {
-                headerIds = headerIds.map(h => headerMap[h] || h);
-
-                // Patch all of the header ids in the user state and copy
-                // over the local progress that doesn't exist in the signed in
-                // user
-                const urls = Object.keys(currentUser.mapProgress);
```

Comment:
I think we should be checking things based on which urls we've visited as local users. Otherwise we might miss something